### ISSUE TITLE
Do not report RV_RETURN_VALUE_IGNORED_NO_SIDE_EFFECT with Mockito.doAnswer()

### ISSUE THREAD

**Issue by peterxcli** [spotbugs/spotbugs#3334]
## Description

```java
SnapshotDiffManager spy = spy(snapshotDiffManager);

doAnswer(invocation -> {
      String[] split = invocation.getArgument(0, String.class).split("/");
      String keyName = split[split.length - 1];
      return Integer.parseInt(keyName.substring(3)) % 2 == 0;
    }
).when(spy).isKeyInBucket(anyString(), anyMap(), anyString());
```

This code leads to:
```
M D RV: Return value of SnapshotDiffManager.isKeyInBucket(String, Map, String) ignored, but method has no side effect  At TestSnapshotDiffManager.java:[line 689]
```

In a `Mockito.doAnswer(...)` chained call it is OK to ignore the return value.

## Related link

- https://github.com/spotbugs/spotbugs/issues/872
- https://github.com/spotbugs/spotbugs/pull/2550

**Comment by MateuszStefek** [comment]
The same for `Mockito.doReturn()`:

`Mockito.doReturn(aReturnValue).when(receiver).someMethod()`

I guess the solution for #2550 needs to be extended.

**Comment by peterxcli** [comment]
Thanks for this fix!
